src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityEventSummary.kt

```diff
@@ -62,6 +174,128 @@ object HoppityEventSummary {
         checkEnded()
     }
 
+    private fun isInInventory(): Boolean =
+        Minecraft.getMinecraft().currentScreen is GuiInventory ||
+            Minecraft.getMinecraft().currentScreen is GuiChest
+
+    private fun reCheckInventoryState() {
+        if (isInInventory() != lastKnownInInvState) {
+            lastKnownInInvState = !lastKnownInInvState
+            lastKnownStatHash = 0
+        }
+    }
+
+    private fun handleResetRequest(args: Array<String>) {
+        // Send a confirmation message confirming this is destructive
+        if (args.any { it.equals("confirm", ignoreCase = true) }) {
+            resetStats()
+            return
+        }
+        ChatUtils.clickableChat(
+            "§c§lWARNING! §r§7This will reset all Hoppity Event stats for all years. " +
+                "Click here or type §c/shresethoppityeventstats confirm §7to confirm.",
+            onClick = ::resetStats
+        )
+    }
+
+    private fun resetStats() {
+        ProfileStorageData.profileSpecific?.let {
```

Comment:
replace all the calls to ProfileStorageData with a val with a getter for storage